非同期通信機能の実装

Daisekkei opened the pull request:
#What
Chat Spaceに対して、送信メッセージ、画像に非同期通信機能を実装

#Why
メッセージ送信後、タイムリーに送信内容をメッセージ画面へ表示するため

検証動画
https://s27.aconvert.com/convert/p3r68-cdx67/h8tiw-c6fe1.gif

Daisekkei committed:
非同期通信機能の実装

Daisekkei committed:
Update _chat-main.scss

i-date commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,67 @@
+  .fail(function() {
+    alert("メッセージ送信に失敗しました");
+});
+  return false;
```
`return false;`とするのではなく、投稿ボタンに対して`prop('disabled', false)`を使用して連続投稿できるようにしましょう。
その際、doneでもfailでも`$('投稿ボタンのセレクタ').prop('disabled', false);`（投稿ボタンのセレクタは適宜書き換えてください）が必要となりますが、コードが重複するのは避けたいので、いつでも実行する処理を記述する`.always(function(){})`を使用してみましょう。`.always(function(){})`の位置によっては、failの閉じ括弧後の`;`は削除しないとエラーになるので注意しましょう。
```
.done(function(data){
  略
})
.fail(function() {
  略
})    // ここの行末に ; があるとエラーになります
.always(function() {
  $('投稿ボタンのセレクタ').prop('disabled', false);
});
```

Daisekkei replied:
ありがとうございます。
message.js
の最後を下記の様に修正し、動作確認いたしました。

})
.always(function() {
  $('.message-form__send-btn').prop('disabled', false);

i-date commented on app/views/layouts/application.html.haml:
```diff
@@ -4,8 +4,10 @@
-    = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload'
-    = javascript_include_tag 'application', 'data-turbolinks-track': 'reload'
+    -# = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload'
+    -# = javascript_include_tag 'application', 'data-turbolinks-track': 'reload'
```
7-8行目は不要なコメントアウトでしたら削除しましょう。

Daisekkei replied:
削除いたしました

i-date reviewed (commented):
ご提出＆gifありがとうございます！コメントいたしました。

i-date commented:
再提出ありがとうございます！
githubの反映処理が遅いだけかもしれないですが、最新修正分がまだ反映されていないようです。
お手数ですがcommit＆push忘れなどをご確認いただけないでしょうか？

Daisekkei committed:
非同期通信修正

#What
message.jp について
return false;
の記述を
 .always(function()
へ変更
#Why
 連続投稿ができる様にするため

#What
application.html.haml
のコメントアウト部分を削除
#Why
不必要な記述を削除するため

Daisekkei commented:
> 再提出ありがとうございます！
> githubの反映処理が遅いだけかもしれないですが、最新修正分がまだ反映されていないようです。
> お手数ですがcommit＆push忘れなどをご確認いただけないでしょうか？

申し訳ありませんでした。
commit&push
が抜けておりました。
実行しましたので、宜しくお願い致します。

shingoteshima commented:
LGTMです！！
